megamek/src/megamek/client/bot/princess/BotGeometry.java

```diff
@@ -158,7 +158,8 @@ public int judgeArea(ConvexBoardArea a) {
             boolean flip = getDirection() > 2;
             HexLine[] edges = a.getEdges();
             if ((edges[getDirection()] == null) || (edges[(getDirection() + 3) % 6] == null)) {
-                LogManager.getLogger().error(new IllegalStateException("Detection of NULL edges in ConvexBoardArea: " + a));
+                LogManager.getLogger().error("Detection of NULL edges in ConvexBoardArea: " + a,
+                        new IllegalStateException());
```

Comment:
Because of how often this function gets called (with a big O of #paths for moving unit * #enemies for each moving unit), I'm really against initializing unnecessary throwaway objects here, even if it only happens in a fraction of the calls to the function. I get that the logger derives some information from the exception, but in this case, performance trumps pretty much every other consideration.

Reply:
Done